osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -199,6 +204,9 @@ protected virtual void AddNested(DrawableHitObject h)
         /// <param name="application">The callback that applies changes to the <see cref="JudgementResult"/>.</param>
         protected void ApplyResult(Action<JudgementResult> application)
         {
+            if (HasFailed.Value)
```

Comment:
This shouldn't be here.

Reply:
To block the object from applying changes to the judgement result, so the drawable hit circle for example doesn't show the explosion effect